drivers/hmis/spec/system/hmis/assessment_definitions_spec.rb

```diff
@@ -0,0 +1,171 @@
+        click_button 'New Assessment'
+        click_link definition.title
+        mui_date_select 'Assessment Date', date: today
+
+        expect do
+          click_button 'Submit'
+          assert_text "#{c1.brief_name} Assessments" # wait until we're back on the assessment table
+        end.to change(e1.custom_assessments, :count).by(1).
+          and change(definition.form_processors, :count).by(1)
+      end
+    end
+
+    context 'with an existing assessment that was created with an old definition' do
+      let!(:assessment) { create(:hmis_custom_assessment, definition: old_definition, enrollment: e1, assessment_date: today) }
+
+      it 'opens the assessment with the original form version' do
+        click_link old_definition.title
+
+        assert_text old_definition.title
+        assert_text 'Assessment Date'
+        assert_text assessment.assessment_date.strftime('%m/%d/%Y')
+        assert_text 'Text on old form'
+      end
+
+      it 'upgrades to the newer version when you unlock for editing' do
+        click_link old_definition.title
+
+        assert_text old_definition.title
+
+        # unlock the assessment should upgrade to the newer form
+        click_button('Unlock Assessment', match: :first)
+        assert_no_text old_definition.title
+        assert_text definition.title
+        expect(page).to have_field('Assessment Date', with: assessment.assessment_date.strftime('%m/%d/%Y'))
+        assert_no_text 'Text on old form'
+      end
+    end
+
+    context 'with an existing WIP assessment that was created with an old definition' do
+      let!(:wip_assessment) { create(:hmis_custom_assessment, definition: old_definition, wip: true, values: { 'assessment_date': today.strftime('%Y-%m-%d') }, enrollment: e1, assessment_date: today) }
+
+      it 'uses old form version for editing' do
+        click_link old_definition.title
+        assert_text old_definition.title
+        expect(page).to have_field('Assessment Date', with: wip_assessment.assessment_date.strftime('%m/%d/%Y'))
+        assert_text 'Text on old form'
+      end
+    end
+  end
+
+  context 'Performing Household Intake assessments with various definition versions' do
+    let!(:definition) { create :hmis_intake_assessment_definition, title: 'New Special Intake' }
+    let!(:old_definition) do
+      fd = create(:hmis_intake_assessment_definition, identifier: definition.identifier, title: 'Old Special Intake', version: 0, status: :retired)
+      fd.definition['item'][0]['item'] << { 'type': 'DISPLAY', 'link_id': 'old_message', 'text': 'Text on old form' }
+      fd.save!
+      fd
+    end
+
+    # e1 (HoH): Intake was Submitted with old form
+    let!(:c1) { create :hmis_hud_client, data_source: ds1, first_name: 'Parent', last_name: 'Jones' }
+    let!(:e1) { create :hmis_hud_wip_enrollment, data_source: ds1, project: p1, client: c1, entry_date: 2.months.ago, relationship_to_hoh: 1 }
```

Comment:
Maybe, change `e1` and `e2` to use `hmis_hud_enrollment` factory (rather than `wip`) since they both have completed intakes already?

Reply:
yes that was an accident, thanks!